mono/metadata/object-internals.h

```diff
@@ -631,7 +631,7 @@ typedef struct {
 	gpointer (*create_delegate_trampoline) (MonoDomain *domain, MonoClass *klass);
 	gpointer (*interp_get_remoting_invoke) (gpointer imethod, MonoError *error);
 	GHashTable *(*get_weak_field_indexes) (MonoImage *image);
-#ifdef TARGET_OSX
+#ifndef HOST_WIN32
```

Comment:
Should we just have the same data structures for all platforms where easy? Even if we don't use the  fields?